crates/bevy_asset/src/handle.rs

```diff
@@ -122,7 +122,7 @@ impl std::fmt::Debug for StrongHandle {
 ///
 /// [`Handle::Strong`] also provides access to useful [`Asset`] metadata, such as the [`AssetPath`] (if it exists).
 #[derive(Component, Reflect)]
-#[reflect(Component)]
+#[reflect(Default, Component)]
```

Comment:
Not 100% necessary but we could also register `Debug`, `Hash`, and `PartialEq`

Reply:
I've been told not to implement unnecessary traits before to avoid code and metadata bloat. Maybe bevy should include this kind of thing in a coding guide to avoid inconsistent messaging?

Reply:
Sorry, to clarify, I don't mean deriving `Debug`, `Hash`, and `PartialEq`. They're actually already implemented. I'm suggesting we register these traits as well:

```suggestion
#[reflect(Default, Component, Debug, Hash, PartialEq)]
```

Reply:
Sure, but that also adds some bloat. Happy to add it though.

Reply:
I wouldn't necessarily call it bloat. It only adds a handful of lines per trait to the generated output IIRC